### ISSUE TITLE
How to return an error in Request Transform 

### ISSUE THREAD

**Issue by bppn** [microsoft/reverse-proxy#2038]
We have written a custom transform provider to parse and patch the body of post requests.

However when there is an invalid request body, the request should immediatly fail.

How can we return a response from the RequestTransform?

Example:

```c#
internal class TransformRequest : ITransformProvider
{

	public void ValidateCluster(TransformClusterValidationContext context)
	{

	}

	public void ValidateRoute(TransformRouteValidationContext context)
	{

	}

	public void Apply(TransformBuilderContext transformBuildContext)
	{

		transformBuildContext.AddRequestTransform(async transformContext =>
		{
			using var reader = new StreamReader(transformContext.HttpContext.Request.Body);
			var body = await reader.ReadToEndAsync();
			if (!string.IsNullOrEmpty(body))
			{

				// Process request body and do tranform

				// --> Error detected in request body -> how to return error?


				var bytes = Encoding.UTF8.GetBytes(body);
				// Change Content-Length to match the modified body, or remove it.
				transformContext.HttpContext.Request.Body = new MemoryStream(bytes);
				// Request headers are copied before transforms are invoked, update any needed headers on the ProxyRequest
				transformContext.ProxyRequest.Content.Headers.ContentLength = bytes.Length;
			}
		});
	}
}
```

**Comment by MihaZupan** [comment]
If you use [YARP 2.0.0 (released yesterday)](https://github.com/microsoft/reverse-proxy/releases), you can write the error to the `HttpContext.Response` directly from inside the request transform.

If we see that you've set the status code on the response, or started writing to the response body, we'll short circuit and stop the forwarding operation.

**Comment by bppn** [comment]
Owkay, thanks for the quick response.  I was indeed using 1.1 and waiting for the release.

**Comment by karelz** [comment]
Triage: Fixed in YARP 2.0. Closing.
